pkg/operator/operatorloglevel/loglevel_controller.go

```diff
@@ -0,0 +1,187 @@
+		},
+		{
+			Group:    "operator.openshift.io",
+			Version:  "v1",
+			Resource: "cloudcredentials",
+		},
+		{
+			Group:    "operator.openshift.io",
+			Version:  "v1",
+			Resource: "kubestorageversionmigrators",
+		},
+		{
+			Group:    "operator.openshift.io",
+			Version:  "v1",
+			Resource: "authentications",
+		},
+		{
+			Group:    "operator.openshift.io",
+			Version:  "v1",
+			Resource: "openshiftcontrollermanagers",
+		},
+		{
+			Group:    "operator.openshift.io",
+			Version:  "v1",
+			Resource: "storages",
+		},
+		{
+			Group:    "operator.openshift.io",
+			Version:  "v1",
+			Resource: "networks",
+		},
+		{
+			Group:    "operator.openshift.io",
+			Version:  "v1",
+			Resource: "consoles",
+		},
+		{
+			Group:    "operator.openshift.io",
+			Version:  "v1",
+			Resource: "csisnapshotcontrollers",
+		},
+		{
+			Group:    "operator.openshift.io",
+			Version:  "v1",
+			Resource: "clustercsidrivers",
+		},
+		{
+			Group:    "operator.openshift.io",
+			Version:  "v1",
+			Resource: "servicecas",
+		},
+	}
+
+	for _, gvr := range gvrs {
+		customResources, err := c.dynamicClient.Resource(gvr).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			klog.V(4).Infof("error trying to list custom resources for %s: %v", gvr.Resource, err)
+			continue
+		}
+
+		for _, cr := range customResources.Items {
+			crCopy := cr.DeepCopy()
+			eventMsgs, needsUpdate := normalizeLogLevelField(crCopy, gvr.Resource)
+			if needsUpdate {
+				if _, err := c.dynamicClient.Resource(gvr).Update(ctx, crCopy, metav1.UpdateOptions{}); err != nil {
+					klog.Warningf("failed to normalize log level to %v for operator %s: %v", operatorv1.Normal, gvr.Resource, err)
+					continue
+				}
+				for _, event := range eventMsgs {
+					syncCtx.Recorder().Event("OperatorLogLevelChange", event)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func normalizeLogLevelField(cr *unstructured.Unstructured,
+	resourceName string) ([]string, bool) {
+	needsUpdate := false
+	eventMsgs := []string{}
+
+	for _, logLevelFieldPath := range [][]string{{"spec", "operatorLogLevel"}, {"spec", "logLevel"}} {
+		// custom resources that do not have log level field are ignored.
+		currentLogLevel, ok, err := unstructured.NestedString(cr.UnstructuredContent(), logLevelFieldPath...)
+		if err != nil {
+			klog.V(4).Infof("failed to find %q in custom resource %s: %v", strings.Join(logLevelFieldPath, "."), resourceName, err)
+			continue
+		}
+		if !ok {
+			continue
+		}
+
+		if loglevel.ValidLogLevel(operatorv1.LogLevel(currentLogLevel)) {
+			continue
+		}
+
+		if err := unstructured.SetNestedField(cr.UnstructuredContent(), string(operatorv1.Normal), logLevelFieldPath...); err != nil {
+			klog.Warningf("failed to set log level to %s in resource %s", operatorv1.Normal, resourceName)
+			continue
+		}
+
+		eventMsgs = append(eventMsgs, fmt.Sprintf("%q changed from %q to %q", strings.Join(logLevelFieldPath, "."), currentLogLevel, operatorv1.Normal))
```

Comment:
A very minor suggestion, maybe it is better to include resourceName in the event message to show which CR :)